src/interfaces/records/messages/records-write.ts

```diff
@@ -78,15 +79,19 @@ export class RecordsWrite extends Message {
   public static async create(options: RecordsWriteOptions): Promise<RecordsWrite> {
     const currentTime = getCurrentTimeInHighPrecision();
 
-    const dataCid = await computeDagPbCid(options.data);
+    if (options.data !== undefined && options.dataCid !== undefined) {
+      throw new Error('only one parameter between `data` and `dataCid` is allowed');
+    }
+    const dataCid = options.dataCid ?? await computeDagPbCid(options.data);
```

Comment:
is it an issue that above you only check for `undefined` but here you also check for `null`?

Reply:
I think it's fine.